### DISCUSSION TITLE
Change pitch and energy of fastspeech 2 model during inference

### DISCUSSION THREAD

**Discussion by nellorebhanuteja** [espnet/espnet#5221] [General]
Hi

I can change duration output file by changing speed_control_alpha parameter in espnet2.bin.tts_inference.Text2Speech.from_pretrained.

How can I change pitch and energy values?

I am using fastspeech2 model.

**Comment by iamanigeeit** [comment]
> How can I change pitch and energy values?

You can change the generator inference function to include `p_factor` and `e_factor `, and add them to `p_outs` and `e_outs` before `return`
